pyat/at/physics/fastring.py

```diff
@@ -100,3 +101,135 @@ def fast_ring(ring: Lattice, split_inds=[]) -> Tuple[Lattice, Lattice]:
+            * U0 - energy loss [eV] (positive number)
+            * Vrf - RF Voltage set point [V]
+            * Qpx, Qpy - linear chromaticities
+            * A1, A2, A3 - amplitude detuning coefficients
+            * emit_x, emit_y, sigma_dp - equilibrium values [m.rad, m.rad, -]
+
+    Returns:
+        ring (Lattice):    Simple ring
+    """
+
+    # parse everything first
+
+    circumference = ring_dictionary['circumference']
+    harmonic_number = ring_dictionary['harmonic_number']
+
+    energy = ring_dictionary['energy']
+
+    alpha_x = ring_dictionary['alpha_x']
+    alpha_y = ring_dictionary['alpha_y']
+
+    beta_x = ring_dictionary['beta_x']
+    beta_y = ring_dictionary['beta_y']
+
+    Qx = ring_dictionary['Qx']
+    Qy = ring_dictionary['Qy']
+
+    alpha = ring_dictionary['alpha']
+    U0 = ring_dictionary['U0']
+    Vrf = ring_dictionary['Vrf']
+
+    Qpx = ring_dictionary['Qpx']
+    Qpy = ring_dictionary['Qpy']
+
+    A1 = ring_dictionary['A1']
+    A2 = ring_dictionary['A2']
+    A3 = ring_dictionary['A3']
+
+    emit_x = ring_dictionary['emit_x']
+    emit_y = ring_dictionary['emit_y']
+    sigma_dp = ring_dictionary['sigma_dp']
+
+    tau = ring_dictionary['tau']
+    
+    # compute rf frequency
+    frf = harmonic_number * clight / circumference
+
+    # compute slip factor
+    gamma = energy / e_mass
+    eta = alpha-1/gamma**2
+
+    # compute the synchronous phase and the TimeLag
+    phi_s = numpy.arcsin(U0/Vrf)
+    TimeLag = clight * phi_s / (2 * numpy.pi * frf)
+    # generate rf cavity element
+    rfcav = RFCavity('RFC', 0, Vrf, frf, harmonic_number, energy,
+                     TimeLag=TimeLag)
+
+    # Now we will use the optics parameters to compute the uncoupled M66 matrix
+    def sincos(x):
+        return numpy.sin(x), numpy.cos(x)
+
+    s_dphi_x, c_dphi_x = sincos(2*numpy.pi*Qx)
+    s_dphi_y, c_dphi_y = sincos(2*numpy.pi*Qy)
+
+    M00 = c_dphi_x + alpha_x * s_dphi_x
+    M01 = beta_x * s_dphi_x
+    M10 = -(1. + alpha_x**2) / beta_x * s_dphi_x
+    M11 = c_dphi_x - alpha_x * s_dphi_x
+
+    M22 = c_dphi_y + alpha_y * s_dphi_y
+    M23 = beta_y * s_dphi_y
+    M32 = -(1. + alpha_y**2) / beta_y * s_dphi_y
+    M33 = c_dphi_y - alpha_y * s_dphi_y
+
+    M44 = 1.
+    M45 = 0.
+    M54 = eta*circumference
+    M55 = 1
+
+    Mat66 = numpy.array([[M00, M01, 0., 0., 0., 0.],
+                         [M10, M11, 0., 0., 0., 0.],
+                         [0., 0., M22, M23, 0., 0.],
+                         [0., 0., M32, M33, 0., 0.],
+                         [0., 0., 0., 0., M44, M45],
+                         [0., 0., 0., 0., M54, M55]], order='F')
+
+    # generate the linear tracking element, we set a length
+    # which is needed to give the lattice object the correct length
+    # (although it is not used)
+    lin_elem = M66('Linear', m66=Mat66, Length=circumference)
+
+    # Generate the simple quantum diffusion element
+    quantdiff = SimpleQuantDiff('SQD', emit_x, emit_y, sigma_dp,
```

Comment:
Since you are using dictionaries why not passing the dictionary to your new element and decoding it inside the __init__ method

The attribute declaration could then be made simpler:

```
for k,v in dictionnary:
    setattr(self, k, v)
```

Of course you have to extract only the needed values from your ring_dictionnary

It could be interesting to set default values for other parameters if they are not provided, for instance the detuning coeffcients=0, beta=1, alpha=0, such that the user may generate a standard ring with minimal inputs